Fixes RHICOMPL-86: SystemPolicyCard ref_id truncates

akofink opened the pull request:
![image](https://user-images.githubusercontent.com/761923/55010332-65af2800-4fba-11e9-83c0-88c04fd5976f.png)

![image](https://user-images.githubusercontent.com/761923/55011034-af4c4280-4fbb-11e9-87ad-b22e7c6df118.png)

https://projects.engineering.redhat.com/browse/RHICOMPL-86

karelhala commented on src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js:
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { CheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
+import Truncate from 'react-truncate';
```
Isn't this essentially the same as our long text truncate? [/src/PresentationalComponents/LongTextTooltip/LongTextTooltip.js](https://github.com/RedHatInsights/insights-frontend-components/blob/master/src/PresentationalComponents/LongTextTooltip/LongTextTooltip.js)

akofink replied:
This isn't a tooltip. Is that component misnamed or something?

ryelo replied:
I think @karelhala meant to link: https://github.com/RedHatInsights/insights-frontend-components/blob/master/doc/components/truncate.md

akofink replied:
Is there a way to use auto length? How am I supposed to know how many characters? That depends on the width of the browser, right?

ryelo replied:
That's true. We may want to go back and revisit that issue with the other component. This should be fine for now.

karelhala commented on src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js:
```diff
@@ -11,10 +12,16 @@ import {
-        const { policy: { compliant }} = this.props;
+    constructor(policy) {
+        super();
+        this.policy = policy.policy;
```
Please do not assign to class attributes, but use state. With this approach our component will not be re-rendered once policy is changed.

akofink commented:
Updated to only use `this.state`

akofink commented:
What if it wraps instead of overflows? Is that okay?

karelhala commented on src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js:
```diff
@@ -11,10 +12,15 @@ import {
-    complianceIcon = () => {
-        const { policy: { compliant }} = this.props;
+    constructor(policy) {
+        super();
```
You should pass the policy to parent as well. `super(policy)`.

karelhala commented on src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js:
```diff
@@ -11,10 +12,15 @@ import {
+    constructor(policy) {
+        super();
+        this.state = { refIdTruncated: <Truncate lines={ 1 }>{ policy.policy.ref_id }</Truncate>, ...policy };
+        this.onMouseover = this.onMouseover.bind(this);
```
No need to use binds.

karelhala commented on src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js:
```diff
@@ -23,29 +29,39 @@ class SystemPolicyCard extends React.Component {
             </div>;
     }
 
+    onMouseover() {
```
You should use arrow function to automatically bind `this`
```JS
onMouseOver = () => {...}
```

karelhala commented on src/SmartComponents/Inventory/applications/compliance/SystemPolicyCard.js:
```diff
@@ -23,29 +29,39 @@ class SystemPolicyCard extends React.Component {
+        this.setState({ refIdTruncated: this.state.policy.ref_id });
+    }
+
+    onMouseout() {
```
You should use arrow function to automatically bind `this`
```JS
onMouseout= () => {...}
```

akofink commented:
Updated to remove binds and pass policy to `super()`. I also refactored a card in our frontend in a similar way: https://github.com/RedHatInsights/compliance-frontend/pull/113

karelhala commented:
Except for the truncate thing and @ryelo's request looks good! Thank you.

akofink commented:
Updated to wrap the untruncated text:

![image](https://user-images.githubusercontent.com/761923/55343366-e9fd2180-5478-11e9-84c4-f956a37b5851.png)

karelhala commented:
:tada: This PR is included in version 3.41.5 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@red-hat-insights/insights-frontend-components)
- [GitHub release](https://github.com/RedHatInsights/insights-frontend-components/releases/tag/v3.41.5)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
